sw/device/tests/kmac_interrupt_test.c

```diff
@@ -0,0 +1,455 @@
+      kmac_irq = kDifKmacIrqFifoEmpty;
+      LOG_INFO("FifoEmpty");
+      fifo_empty_irq = true;
+      break;
+    case kTopEarlgreyPlicIrqIdKmacKmacDone:
+      LOG_INFO("KmacDone");
+      kmac_irq = kDifKmacIrqKmacDone;
+      kmac_done_irq = true;
+      break;
+    case kTopEarlgreyPlicIrqIdKmacKmacErr:
+      kmac_irq = kDifKmacIrqKmacErr;
+      LOG_FATAL("KmacError!");
+      test_status_set(kTestStatusFailed);
+      break;
+    default:
+      LOG_FATAL("ISR is not implemented!");
+      test_status_set(kTestStatusFailed);
+  }
+
+  CHECK_DIF_OK(dif_kmac_irq_acknowledge(&kmac, kmac_irq));
+}
+
+/**
+ * External interrupt handler
+ *
+ * Handles all peripheral interrupts on Ibex. PLIC asserts an external interrupt
+ * line to the CPU, which results in a call to this handler. This handler
+ * overrides the default implementation, and prototype for this handler must
+ * include appropriate attributes.
+ */
+void handler_irq_external(void) {
+  // Claim the IRQ by reading the Ibex specific CC register.
+  dif_rv_plic_irq_id_t interrupt_id = 0;
+
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic0, kPlicTarget, &interrupt_id));
+
+  // Check if the interrupted peripheral is KMAC.
+  top_earlgrey_plic_peripheral_t peripheral_id =
+      top_earlgrey_plic_interrupt_for_peripheral[interrupt_id];
+  CHECK(peripheral_id == kTopEarlgreyPlicPeripheralKmac,
+        "ISR interrupted peripheral is not KMAC!");
+  handle_kmac_isr(interrupt_id);
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
+  // register.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic0, kPlicTarget, interrupt_id));
+}
+
+/**
+ * Configures all the relevant interrupts in KMAC.
+ */
+static void kmac_configure_irqs(dif_kmac_t *kmac) {
+  CHECK_DIF_OK(
+      dif_kmac_irq_set_enabled(kmac, kDifKmacIrqKmacDone, kDifToggleEnabled));
+  CHECK_DIF_OK(
+      dif_kmac_irq_set_enabled(kmac, kDifKmacIrqFifoEmpty, kDifToggleEnabled));
+}
+
+/**
+ * Configures all the relevant interrupts in PLIC.
+ */
+static void plic_configure_irqs(dif_rv_plic_t *plic) {
+  // Set IRQ priorities to MAX.
+  CHECK_DIF_OK(dif_rv_plic_irq_set_priority(
+      plic, kTopEarlgreyPlicIrqIdKmacKmacDone, kDifRvPlicMaxPriority));
+
+  CHECK_DIF_OK(dif_rv_plic_irq_set_priority(
+      plic, kTopEarlgreyPlicIrqIdKmacFifoEmpty, kDifRvPlicMaxPriority));
+
+  // Set Ibex IRQ priority threshold level.
+  CHECK_DIF_OK(dif_rv_plic_target_set_threshold(&plic0, kPlicTarget,
+                                                kDifRvPlicMinPriority));
+
+  // Enable IRQs in PLIC.
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+      plic, kTopEarlgreyPlicIrqIdKmacKmacDone, kPlicTarget, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(plic,
+                                           kTopEarlgreyPlicIrqIdKmacFifoEmpty,
+                                           kPlicTarget, kDifToggleEnabled));
+}
+
+/**
+ * @brief  Run SHA-3 test cases using single blocking absorb/squeeze operations.
```

Comment:
Nit: Inconsistent with the above function documentation.
Nit2: We normally use the following format:
```
/**
 * Runs SHA-3 test cases using single blocking absorb/squeeze operations.
 * /
```